tests/components/tcp/test_sensor.py

```diff
@@ -57,6 +57,18 @@ def mock_select_fixture():
         yield mock_select
 
 
+@pytest.fixture(name="mock_ssl_context")
+def mock_ssl_context_fixture():
+    """Mock select."""
+    with patch(
+        "ssl.create_default_context",
```

Comment:
Can you change the patch target to just patch ssl in the integration instead of the global one?

Reply:
sure but why, out of curiosity? As patch is restored within the context manager (after the test) and shouldn't affect threaded tests either?
It's clearer to see what's being mocked when patching the original package

Reply:
We try to avoid broad patch targets because with asyncio anything in the loop can happen while the patch is in place which makes it hard to debug tests when the CI is failing intermittently. While it's unlikely in this case that something already running is going to need to create and ssl context while the patch is in place, we treat this as a best practice to avoid the problem in the first place.

Reply:
makes sense, not really familiar with asyncio (even python3 is too recent for my old python :trollface: )
I'd expect pytest/unittest to properly segregate event loops for tests to avoid mocks and monkeypatches commonly done in these scenarios..
but definitely better to be safe than sorry. change done :heavy_check_mark: